### ISSUE TITLE
en-gb ordinal formatted number has surrounding square brackets

### ISSUE THREAD

**Issue by jwarby** [iamkun/dayjs#1892]
**Describe the bug**
I'm not sure if this is intentional or not, but when using `dayjs.localeData().ordinal(2)` with `en-gb` locale, the result string is wrapped in square brackets, eg `"[2nd]"`.

**Expected behavior**
The ordinal number to be returned sans square brackets, eg `"2nd"`

**Information**
 - Day.js Version: v1.11.2
 - OS: Ubuntu 20.04
 - Browser: Chrome 99
 - Time zone: GMT+01:00 BST (British Summer Time)

**Comment by BePo65** [comment]
At the moment the `dayjs.localeData().ordinal(..)` value is only used by the `advancedFormat` plugin and here the square brackets are interpreted as 'escape characters' and are automatically removed.

But to be consistent with all other locale plugins, the square brackets should be removed. I will create a pr for this.

**Comment by jwarby** [comment]
Thanks for looking into this @BePo65 and sorry for my late response back, I didn't seem to get a notification from GitHub about this.

What you've said makes sense, and I've actually hit the issue you were describing with the workaround I had applied. I think this just stems from me trying to (ab)use Day.js to format a plain number as an ordinal, so I've changed my approach to use [`react-intl`'s `selectordinal` formatting](https://formatjs.io/docs/core-concepts/icu-syntax/#selectordinal-format) instead for this particular use case (just mentioning that here in case anybody else comes across this issue and is trying to do the same thing as me).

Thanks again!
